performances/performance_test/test/test_system.cpp

```diff
@@ -32,12 +32,12 @@ TEST_F(TestSystem, SystemAddNodesTest)
     std::vector<std::shared_ptr<performance_test::Node>> nodes_vec = {node_2, node_3};
 
     auto system_executor = performance_test::STATIC_SINGLE_THREADED_EXECUTOR;
-    performance_test::System separate_threads_system(system_executor);
+    performance_test::System<performance_test::Node> separate_threads_system(system_executor);
```

Comment:
regardless of how we end up implementing this new toggle, we shouldn't have to change the signature of these function, but rather make sure that the old code still compiles and behaves as before.

Reply:
I added the `performance_test::Node` as the default template argument, but the syntax will still need to be like this: `performance_test::System<>` in order to compile